pkg/controller/volume/cache/desired_state_of_world.go

```diff
@@ -142,14 +146,15 @@ type volumeToAttach struct {
 	// volume and are scheduled to the underlying node. The key in the map is
 	// the name of the pod and the value is a pod object containing more
 	// information about the pod.
-	scheduledPods map[types.UniquePodName]pod
+	scheduledPods map[types.UniquePodName]Pod
 }
 
-// The pod object represents a pod that references the underlying volume and is
+// Pod represents a pod that references the underlying volume and is
 // scheduled to the underlying node.
-type pod struct {
-	// podName contains the name of this pod.
-	podName types.UniquePodName
+type Pod struct {
```

Comment:
Leave the old `type pod struct` as is. Create a new externally visible struct `type PodToAttach struct` this struct can contain any fields that your populator will need (e.g. PodName, PodUID, VolumeName, NodeName, etc.) and `GetPods()` can construct and return this object on the fly using the internal `node`/`volume`/`pod` objects.